rudder-core/src/main/scala/com/normation/rudder/services/nodes/NodeInfoService.scala

```diff
@@ -111,84 +111,97 @@ object NodeInfoServiceImpl {
-  , ldapMapper         : LDAPEntityMapper
-  , inventoryMapper    : InventoryMapper
-  , inventoryDitService: InventoryDitService
-) extends NodeInfoService with Loggable {
-  import NodeInfoServiceImpl._
 
-  def getNode(nodeId: NodeId): Box[Node] = {
-    for {
-      con <- ldap
-      entry <- con.get(nodeDit.NODES.NODE.dn(nodeId.value), nodeInfoAttributes:_*) ?~! s"Node with ID '${nodeId.value}' was not found"
-      node <- ldapMapper.entryToNode(entry)
-    } yield {
-      node
-    }
-  }
+/**
+ * A cache on top of node info service.
+ *
+ */
 
-  def getLDAPNodeInfo(nodeId: NodeId) : Box[LDAPNodeInfo] = {
-    logger.trace("Fetching node info for node id %s".format(nodeId.value))
-    for {
-      con <- ldap
-      node <- con.get(nodeDit.NODES.NODE.dn(nodeId.value), nodeInfoAttributes:_*) ?~! "Node with ID '%s' was not found".format(nodeId.value)
-      nodeInventory  <- con.get(inventoryDit.NODES.NODE.dn(nodeId.value), nodeInfoAttributes:_*)  ?~! "Node info with ID '%s' was not found".format(nodeId.value)
-      machineInventory <- {
-                            val machines = inventoryMapper.mapSeqStringToMachineIdAndStatus(nodeInventory.valuesFor(A_CONTAINER_DN))
-                            if(machines.size == 0) Full(None)
-                            else { //here, we only process the first machine
-                              if(logger.isDebugEnabled && machines.size > 1) {
-                                logger.debug("Node with id %s is attached to several container. Taking %s as the valid one, ignoring other (%s)".format(
-                                    nodeId.value, machines(0)._1.value, machines.tail.map( _._1).mkString(", "))
-                                )
-                              }
-                              val (machineId, status) = machines(0)
-                              val dit = inventoryDitService.getDit(status)
-
-                              Full(con.get(dit.MACHINES.MACHINE.dn(machineId),Seq("objectClass"):_*).toOption)
-                            }
-                          }
-    } yield {
-      LDAPNodeInfo(node, nodeInventory, machineInventory.map( _.valuesFor("objectClass").toSeq))
-    }
-  }
+class NodeInfoServiceCachedImpl(
+    ldap           : LDAPConnectionProvider[RoLDAPConnection]
+  , nodeDit        : NodeDit
+  , inventoryDit   : InventoryDit
+  , ldapMapper     : LDAPEntityMapper
+) extends NodeInfoService with Loggable {
 
 
-  def getNodeInfo(nodeId: NodeId) : Box[NodeInfo] = {
-    for {
-      ldapEntries <- getLDAPNodeInfo(nodeId)
-      nodeInfo    <- ldapMapper.convertEntriesToNodeInfos(
-                         ldapEntries.nodeEntry
-                       , ldapEntries.nodeInventoryEntry
-                       , ldapEntries.machineObjectClasses
-                     )
-    } yield {
-      nodeInfo
-    }
-  }
+  /*
+   * Our cache
+   */
+  private[this] var nodeCache = Option.empty[Map[NodeId, (LDAPNodeInfo, NodeInfo, Node)]]
```

Comment:
It's exactly the same as:
private[this] var nodeCache: Option[Map[NodeId, (LDAPNodeInfo, NodeInfo, Node)]] = None